app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -162,6 +165,27 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
     }
   }
 
+  def showHaveYourNationalInsuranceNumber(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
+    viewConfig.isLoggedIn.flatMap { isLoggedIn: Boolean =>
+      Future.successful(Ok(do_you_have_your_nino(DoYouHaveYourNino.apply(), completionUrl, isLoggedIn, failureUrl)))
+    }
+  }
+
+  def processHaveYourNationalInsuranceNumber(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
+    viewConfig.isLoggedIn.flatMap { isLoggedIn: Boolean =>
+      DoYouHaveYourNino.apply().bindFromRequest().fold(
+        errors =>
+          Future.successful(BadRequest(do_you_have_your_nino(
+            errors, completionUrl, isLoggedIn, failureUrl
+          ))),
+        {
+          case UserHasNinoTrue => Future.successful(Redirect(routes.PersonalDetailsCollectionController.whatIsYourNino(completionUrl, failureUrl)).withSession(Session(request.session.data ++ Map(HAS_NINO_KEY -> "yes"))))
+          case UserHasNinoFalse => Future.successful(Redirect(routes.PersonalDetailsCollectionController.whatIsYourNino(completionUrl, failureUrl)).withSession(Session(request.session.data ++ Map(HAS_NINO_KEY -> "no"))))
```

Comment:
The cases UserHasNinoTrue and UserHasNinoFalse don't seem to be tested - I could be mistaken

Reply:
As the wiring ticket is not done yet to send them to the correct place for yes or no, I have not added the test for these. What is your nino was selected to get the session storage working as it needed a destination.

Reply:
Fair enough!